allennlp/training/trainer.py

```diff
@@ -627,7 +627,13 @@ def train(self) -> Dict[str, Any]:
         """
         Trains the supplied model with the supplied parameters.
         """
-        epoch_counter, validation_metric_per_epoch = self._restore_checkpoint()
+        try:
+            epoch_counter, validation_metric_per_epoch = self._restore_checkpoint()
+        except RuntimeError:
+            logger.error("Could not continue training from the checkpoint.  Did you mean to output to a different "
```

Comment:
The stack trace from the `raise` below will bury this message.  I'd probably print the stack trace, then raise a new error with this nicer message (also saying that the `restore_checkpoint` stack trace is available above, in case it is needed for some reason).

Reply:
Is it better to exit with a new exception or to log an error message and `sys.exit(-1)`?

Reply:
Probably an exception?  That way it's easier to see in the code where things failed.

Reply:
We would be printing a stack trace on the prior line.

Reply:
Yeah, fair point.  Whichever you think is best is fine.